src/main/java/org/veupathdb/service/eda/compute/plugins/correlationassaymetadata/CorrelationAssayMetadataPlugin.java

```diff
@@ -48,16 +48,48 @@ public List<StreamSpec> getStreamSpecs() {
     // Wrangle into correct types for what follows
     EntityDef entity = getContext().getReferenceMetadata().getEntity(entityId).orElseThrow();
 
-    // Grab all continuous variables from ancestors
+    // Find ancestors
     ReferenceMetadata metadata = getContext().getReferenceMetadata();
-    List<VariableDef> metadataVariables = metadata.getAncestors(entity).stream() // Get all ancestors of entity.
-        .filter(ancestor -> !getManyToOneWithDescendant(metadata, ancestor, entity)) // Filter to those that are one-to-one with target entity or ancestor of entity.
-        .flatMap(entityDef -> entityDef.getVariables().stream()) // Flatten stream of var streams into a single stream of vars.
-        .filter(var -> var.getDataShape() == APIVariableDataShape.CONTINUOUS && var.getSource().isResident()) // Filter out inherited and non-continuous variables.
-        .filter(var -> !var.getVariableId().contains("_stable_id")) // Filter out id variables
-        .collect(Collectors.toList());
-
-    LOG.info("Metadata variables that are one-to-one with parent for ancestors of entity ID: {} -- {}", entity.getId(),
+    List<EntityDef> ancestors = metadata.getAncestors(entity);
+
+    // Grab all metadata variables appropriate for this compute. When there are ancestor entities,
+    // we only want those variables that are on a one-to-one path with the current entity. When there
+    // are no ancestor entities, we'll take all variables that are not part of a collection as metadata variables.
+    List<VariableDef> metadataVariables;
+    if (ancestors.isEmpty()) {
+
+      LOG.info("No ancestors found, using all variables that are not part of a collection.");
+      
+      // Grab all non-id variables from this entity
+      metadataVariables = entity.getVariables().stream()
+          .filter(var -> !var.getVariableId().contains("_stable_id"))
+          .collect(Collectors.toList());
```

Comment:
Very minor, but if you want the filtering to all be together, you could move this below the definition of `variablesInACollection` and add a filter:

```
      metadataVariables = entity.getVariables().stream()
          .filter(var -> !var.getVariableId().contains("_stable_id"))
          .filter(var -> !variablesInACollection.contains(var))
          .collect(Collectors.toList());
```